src/caip-types.ts

```diff
@@ -0,0 +1,165 @@
+import type { Infer } from 'superstruct';
+import {
+  array,
+  is,
+  object,
+  optional,
+  pattern,
+  size,
+  string,
+} from 'superstruct';
+
+export const CHAIN_ID_REGEX =
+  /^(?<namespace>[-a-z0-9]{3,8}):(?<reference>[-a-zA-Z0-9]{1,32})$/u;
+
+export const ACCOUNT_ID_REGEX =
+  /^(?<chainId>(?<namespace>[-a-z0-9]{3,8}):(?<reference>[-a-zA-Z0-9]{1,32})):(?<accountAddress>[a-zA-Z0-9]{1,64})$/u;
```

Comment:
This is inaccurate, per CAIP-10 an account address is up 128 characters, our regex limits to 64 characters.